src/geometry/manifold/smooth_manifold_with_corners.lean

```diff
@@ -619,6 +619,50 @@ instance prod {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
 
 end smooth_manifold_with_corners
 
+namespace topological_space.opens
+
+open topological_space
+
+variables  {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+  {E : Type*} [normed_group E] [normed_space 𝕜 E]
+  {H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+  {M : Type*} [topological_space M] [charted_space H M] [smooth_manifold_with_corners I M]
+  (s : opens M)
+
+instance : smooth_manifold_with_corners I s :=
+{ ..topological_space.opens.has_groupoid (times_cont_diff_groupoid ∞ I) s }
+
+end topological_space.opens
+
+lemma local_homeomorph.singleton_smooth_manifold_with_corners
+  {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+  {E : Type*} [normed_group E] [normed_space 𝕜 E]
+  {H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+  {M : Type*} [topological_space M]
+  (e : local_homeomorph M H) (h : e.source = set.univ) :
+  @smooth_manifold_with_corners 𝕜 _ E _ _ H _ I M _ (e.singleton_charted_space h) :=
+{ compatible := begin
+    intros e' e'' he' he'',
+    rw singleton_charted_space_one_chart e h e' he',
+    rw singleton_charted_space_one_chart e h e'' he'',
+    refine (times_cont_diff_groupoid ∞ I).eq_on_source _ e.trans_symm_self,
+    have hle : id_restr_groupoid ≤ (times_cont_diff_groupoid ∞ I) :=
+      (closed_under_restriction_iff_id_le (times_cont_diff_groupoid ∞ I)).mp (by apply_instance),
+    exact structure_groupoid.le_iff.mp hle _ (id_restr_groupoid_mem _),
+  end }
+/-
+{ ..(e.singleton_has_groupoid h (times_cont_diff_groupoid ∞ I)) } does not work. Why!?
+-/
```

Comment:
Also here, I'm wondering

Reply:
@hrmacbeth do you know why this does not work?

Reply:
Good question, I checked out the branch and played with it but couldn't figure it out.  Maybe worth making a mwe and posting on Zulip.

For interested readers, the error varies (depending on the precise setup) between
```
failed to synthesize type class instance for
...
⊢ charted_space H M
```
and 
```
synthesized type class instance is not definitionally equal to expression inferred by typing rules, synthesized
  _inst
inferred
  e.singleton_charted_space h
```

Reply:
Asked on Zulip:
https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Type.20search.20problem

Reply:
@Nicknamen Did you see Mario's idea for dealing with this, on Zulip?  Do you want to implement it?

Reply:
Sure!